src/OmnichannelChatSDK.ts

```diff
@@ -1519,13 +1519,43 @@ class OmnichannelChatSDK {
         }
 
         const {protocol} = optionalParams;
-        const supportedChatAdapterProtocols = [ChatAdapterProtocols.ACS, ChatAdapterProtocols.IC3];
+        const supportedChatAdapterProtocols = [ChatAdapterProtocols.ACS, ChatAdapterProtocols.IC3, ChatAdapterProtocols.DirectLine];
         if (protocol && !supportedChatAdapterProtocols.includes(protocol as string)) {
             return Promise.reject(`ChatAdapter for protocol ${protocol} currently not supported`);
         }
 
-        if (protocol === ChatAdapterProtocols.ACS || this.liveChatVersion === LiveChatVersion.V2) {
-            return new Promise (async (resolve, reject) => { // eslint-disable-line no-async-promise-executor
+        if (protocol === ChatAdapterProtocols.DirectLine) {
```

Comment:
Im not a fan of long methods doing more than one thing , I would recommend to break it into small , functionally atomic methods, but is up to you, not a blocker.

Reply:
Refactored